compiler/aarch64/runtime/ARM64arrayCopy.spp

```diff
@@ -211,14 +288,17 @@ bwDstAlign8:
 bwDstAlign16:
 	tst	x1, #8
 	bne	bwDoubleWordCopyLoop		// srcAddr is not 16-byte aligned
-bwQuadWordCopyLoop:
+__bwQuadWordCopyLoop:
+	cbz	x0, finished			// return if no bytes to copy
+	subs	x3, x2, x1
+	beq	finished			// return if srcAddr == dstAddr
 	// Both srcAddr and dstAddr are 16-byte aligned
 	cmp	x0, #16
 	blt	bwDoubleWordCopy		// less than 16 bytes remaining
 	ldp	x3, x4, [x1, #-16]!
 	sub	x0, x0, #16
 	stp	x3, x4, [x2, #-16]!
-	b	bwQuadWordCopyLoop
+	b	__bwQuadWordCopyLoop
```

Comment:
The same comment as the one for `__fwQuadWordCopyLoop`.
The additional checks will slow down this loop.